command/agent/http.go

```diff
@@ -193,6 +194,58 @@ func NewHTTPServers(agent *Agent, config *Config) ([]*HTTPServer, error) {
 		srvs = append(srvs, srv)
 	}
 
+	// Initialize the API socket if configured
+	startAPISocket := func() {
+		if config.APISocket.Path == "" {
+			return
+		}
+
+		const (
+			MaxSocketPathLength = 104
```

Comment:
This varies depending on OS. So I think we should just try to create the socket and return the awful OS error if we fail.

If Go can find the `sun_path` length for the OS its on, that would be amazing. I think even then we need to be careful to subtract 1 to leave space for a null terminator: otherwise the man page implies no one may be able to connect to it?! What a mess. https://man7.org/linux/man-pages/man7/unix.7.html

Reply:
Looks like syscall exposes it: https://pkg.go.dev/syscall#RawSockaddrUnix

Still might need to subtract 1 to leave room for a null terminator?

Reply:
Oh actually maybe you already peeked because it looks like 104 is the floor!

Peeking at what Go sets across platforms:
```
go/src/syscall$ rg -A3 RawSockaddrUnix | rg Path.*int | awk '{print $3}' | sort -u
[1023]uint8
[104]int8
[108]int8
[UNIX_PATH_MAX]int8
```

Somehow AIX manages to be the nice one here with 1023. What a world. `UNIX_PATH_MAX` is Windows only and 108.

Although I'm not sure if that takes the null terminator weirdness into account? So maybe we should set this to 103?

Reply:
Yeah, I'd peeked, but I think you're right about the null terminator. I think we can go for something more OS-specific by changing that line to: 

```go
			MaxSocketPathLength = len(syscall.RawSockaddrUnix{}.Path) - 1 // remove 1 byte for null terminator
```